### PR TITLE
chore: remove unneeded deps

### PR DESCRIPTION
**Pull request by tac0turtle**
remove final tm dep form iavl

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,46 +93,3 @@ exploremem:
...
-
-proto-lint:
-	@$(DOCKER_BUF) check lint --error-format=json
-.PHONY: proto-lint
-
-proto-check-breaking:
-	@$(DOCKER_BUF) check breaking --against-input $(HTTPS_GIT)#branch=master
-.PHONY: proto-check-breaking
-
-proto-gen:
```
Comment (julienrbrt) [inline Makefile:135]: Do you intend to remove this or is it a mistake?

File: Makefile [hunk 2]
```diff
@@ -93,46 +93,3 @@ exploremem:
...
-	rm -f $(CERTSTRAP) $(PROTOBUF) $(GOX) $(GOODMAN)
-	rm -rf /usr/local/include/google/protobuf
-	rm -f /usr/local/bin/protoc
-.PHONY: tooks-clean
-
-###
-# Non Go tools
-###
-
-protoVer=0.10.0
```
Comment (julienrbrt) [inline Makefile:122]: I feel like this is needed (but it could be upgraded to 0.11.0)

File: Makefile (hunk 1 above)
Comment (tac0turtle) [reply Makefile:135]: there is no more proto in this repo.

Comment (julienrbrt) [reply Makefile:135]: Ah ok 👍 I think we can remove the buf config then right?

**Review by julienrbrt — APPROVED** [review]
lgtm, pending https://github.com/cosmos/iavl/pull/604#discussion_r1008846603

**Pull request merged**